contrib/chacha.h

```diff
@@ -122,6 +146,15 @@ inline uint32_t ChaCha<R>::operator()() {
     return block[idx];
 }
 
+template<size_t R>
+inline double ChaCha<R>::gen_double() {
+    // We have a 32-bit integer result. Doubles support 53-bits of precision
+    // (1 bit implied). Use what we have in a single sample (32-bits precision,
+    // lower 21 bits are zero).
+    uint64_t x = this->operator()();
+    return (x << 21) * 0x1.0p-53;
```

Comment:
Hex float literals are only officially supported in the C++17 standard.

Reply:
`1.1102230246251565e-16` resolves to exactly the same value (at least, when printed with 30 decimal places). Shall I just replace the value?

Reply:
The comment above that line states that 'the lower 21 bits are zero'.
I just checked and this is still true for the value you suggest, so I believe it is safe to replace it.
Please go ahead! Thanks!

Reply:
Done